util/src/main/java/io/kubernetes/client/Configmap.java

```diff
@@ -0,0 +1,203 @@
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.ByteBuffer;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import okhttp3.Call;
+import okhttp3.Response;
+
+/** Utility class to create Configmaps */
+public class Configmap {
+
+  private ApiClient apiClient;
+  private CoreV1Api coreClient;
+
+  public Configmap() {
+    this(Configuration.getDefaultApiClient());
+  }
+
+  public Configmap(ApiClient apiClient) {
+    this.apiClient = apiClient;
+    this.coreClient = new CoreV1Api(apiClient);
+  }
+
+  public ApiClient getApiClient() {
+    return apiClient;
+  }
+
+  public void setApiClient(ApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  /**
+   * Method to create a Configmap from a file. The file should contain key-value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param file
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromFile(String namespace, String name, Path file)
+      throws IOException, CreateConfigmapException, ApiException {
+    if (Files.isRegularFile(file)) {
+      try (BufferedReader bufferedReader =
+          new BufferedReader(new FileReader(String.valueOf(file)))) {
+        createFromReader(namespace, name, bufferedReader);
+      }
+    } else {
+      throw new CreateConfigmapException("Target path is a directory");
+    }
+  }
+
+  /**
+   * To create Configmap by passing InputStream of a file which has key value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param stream
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromStream(String namespace, String name, InputStream stream)
+      throws IOException, CreateConfigmapException, ApiException {
+    createFromReader(namespace, name, new InputStreamReader(stream));
+  }
+
+  /**
+   * To creat Configmap by passing string literal. Value should be a string in the form "key=value"
+   *
+   * @param namespace
+   * @param name
+   * @param value
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromString(String namespace, String name, String value)
+      throws IOException, CreateConfigmapException, ApiException {
+    createFromReader(namespace, name, new StringReader(value));
+  }
+
+  /**
+   * To create Configmap by passing Reader of a file which has key value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param reader
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromReader(String namespace, String name, Reader reader)
```

Comment:
consider rename `createFromFile/createFromString/createFromReader` to `create`. the former function names are in golang-fashion, in java the parameters are overridable

Reply:
:) I'm definitely too influenced by Go, this approach sgtm